### PR TITLE
implemented save button

### PR DESCRIPTION
**Pull request by bceacuna**
This implementation only supports desktop

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -45,7 +45,14 @@
                 <script src="lib/bcc_web.js"></script>
                 <script src="lib/bcgs_ford_main.js"></script>
                  <script src="lib/bcgs_ford_modal.js"></script>
-                <script>videojs("player").FordPlayer({debug:true});</script>
+                <script>videojs("player").FordPlayer({
+                        debug:true,                   
+                        showSaveButton: true,
+                        callbackSave: function(video){ 
+                            console.log("save video: " + video.id);
```
Comment (tyard) [inline index.html]: Should be videoId.

File: lib/bcgs_ford_modal.js [hunk 2]
```diff
@@ -152,6 +153,16 @@ TODO: remove instrinsic ratio implementation from index.html and it to the plugi
...
     $detailDiv.find('.bcc-detail-right').html("Show Details &or;");
     $detailDiv.find('.bcc-detail-right').click(onShowHideDetails);
+            
+    //Shows Save button based on player configuration
+    if (bcc.settings.showSaveButton){
+        var $saveButton = $detailDiv.find('.bcc-save-icon');
+        $saveButton.show();   
+        // Adds event handler
+        $saveButton.on("click touchstart", function(){ 
+            bcc.settings.callbackSave(video)
```
Comment (tyard) [inline lib/bcgs_ford_modal.js]: semicolon

Comment (tyard) [inline lib/bcgs_ford_modal.js]: Actually, you want to pass the videoConfig, not the BC video object.

File: index.html (hunk 1 above)
Comment (tyard) [inline index.html]: Captured as a later comment, but it's videoId because this should be the video config, not the video.

Comment (bceacuna) [inline index.html]: Ok, I thought we had to pass the video object. Let me fix this.

File: lib/bcgs_ford_modal.js (hunk 2 above)
Comment (bceacuna) [inline lib/bcgs_ford_modal.js]: I just pushed my changes. Let me know if it is ok to merge.

File: lib/bcgs_ford_modal.js [hunk 3]
```diff
@@ -152,6 +153,16 @@ TODO: remove instrinsic ratio implementation from index.html and it to the plugi
...
     $detailDiv.find('.bcc-detail-right').html("Show Details &or;");
     $detailDiv.find('.bcc-detail-right').click(onShowHideDetails);
+            
+    //Shows Save button based on player configuration
+    if (bcc.settings.showSaveButton){
+        var $saveButton = $detailDiv.find('.bcc-save-icon');
+        $saveButton.show();   
+        // Adds event handler
+        $saveButton.on("click touchstart", function(){ 
+            bcc.settings.callbackSave(bcc.model.getVideo());
```
Comment (tyard) [inline lib/bcgs_ford_modal.js]: This is already stored in videoConfig. This is fine, but it's another call that isn't strictly necessary.

File: lib/bcgs_ford_modal.js [hunk 4]
```diff
@@ -152,6 +153,16 @@ TODO: remove instrinsic ratio implementation from index.html and it to the plugi
     //TODO: implement localization
     $detailDiv.find('.bcc-detail-right').html("Show Details &or;");
     $detailDiv.find('.bcc-detail-right').click(onShowHideDetails);
+            
+    //Shows Save button based on player configuration
+    if (bcc.settings.showSaveButton){
+        var $saveButton = $detailDiv.find('.bcc-save-icon');
```
Comment (tyard) [inline lib/bcgs_ford_modal.js]: Just something to consider going forward -- defining a local variable this far down in the code can cause potential problems (like hoisting) as the code grows in complexity).

Generally, we try to declare all vars at the top of the scope to make it clear what is defined in that scope, even if the assignments don't come till later.

**Comment by tyard** [comment]
ok to merge, or after the changes suggested

File: lib/bcgs_ford_modal.js (hunk 4 above)
Comment (bceacuna) [inline lib/bcgs_ford_modal.js]: hoisting... good catch. I'll fix this

File: lib/bcgs_ford_modal.js (hunk 3 above)
Comment (bceacuna) [inline lib/bcgs_ford_modal.js]: I replaced bcc.model.getVideo() with videoConfig

**Pull request merged**